pkg/networkmanager/app.jsx

```diff
@@ -0,0 +1,174 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2021 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+import '../lib/patternfly/patternfly-cockpit.scss';
+import cockpit from "cockpit";
+import ReactDOM from "react-dom";
+import React from 'react';
+
+import { Button } from "@patternfly/react-core";
+import { ExclamationCircleIcon } from "@patternfly/react-icons";
+
+import { EmptyStatePanel } from "cockpit-components-empty-state.jsx";
+import { ModelContext } from './model-context.jsx';
+import { NetworkInterfacePage } from './network-interface.jsx';
+import { NetworkPage } from './network-main.jsx';
+
+import * as service from 'service.js';
+import { init as initDialogs, NetworkManagerModel } from './interfaces.js';
+import { superuser } from 'superuser';
+
+const _ = cockpit.gettext;
+
+class App extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            interfaces: [],
+            isModelReady: false,
+            nmEnabled: undefined,
+            nmRunning: undefined,
+            path: cockpit.location.path,
+            privileged: false,
+        };
+
+        this.nmService = service.proxy("NetworkManager");
+        this.nmService.wait(() => {
+            this.nmService.addEventListener('changed', () => {
+                this.setState({ nmEnabled: this.nmService.enabled });
+            });
+        });
+
+        const client = cockpit.dbus("org.freedesktop.NetworkManager", { superuser: "try" });
+        // track NM going away or reappearing
+        client.addEventListener("owner", (event, owner) => {
+            this.setState({ nmRunning: (owner !== null) });
+        });
+
+        this.onModelChange = this.onModelChange.bind(this);
+        this.onNavigate = this.onNavigate.bind(this);
+
+        superuser.addEventListener("changed", () => {
+            this.setState({ privileged: superuser.allowed });
+        });
+    }
+
+    componentDidUpdate(_, prevState) {
+        /* When the model initialization is complete load the Jquery rendered dialogs
+         * and update the state accordingly.
+         */
+        if (!prevState.isModelReady && this.state.isModelReady) {
+            this.model.preinit.then(() => {
+                initDialogs(this.model);
```

Comment:
Let's try to make it so that the call to initDialogs can already happen in `init` below.

Reply:
I have added a FIXUP for this.